src/offset_of.rs

```diff
@@ -94,6 +95,29 @@ macro_rules! offset_of {
     }};
 }
 
+#[cfg(feature = "unstable_const")]
+#[macro_export(local_inner_macros)]
+macro_rules! offset_of {
+    ($parent:path, $field:tt) => {{
+        _memoffset__field_check!($parent, $field);
+
+        // Get a base pointer.
+        // No UB here, and the pointer does not dangle, either.
+        let uninit = $crate::mem::MaybeUninit::<$parent>::uninit();
+        unsafe {
+            // This, on the other hand, *is* UB, since we're creating a reference
+            // to uninitialized data.
+            // Unfortunately it's the best we can do at the moment.
+            let base_ref = $crate::mem::transmute::<_, &$parent>(&uninit);
+            let base_u8_ptr = base_ref as *const _ as *const u8;
+
+            let field_u8_ptr = &base_ref.$field as *const _ as *const u8;
```

Comment:
One last thing: maybe also copy the comment from the other impl here that says
```
// This is another reference to uninitialized data.
// Crucially, we know that this will not trigger a deref coercion because
// of the `field_check!` we did above.
```